profiler/lib.profiler/src/org/netbeans/lib/profiler/heap/ClassDumpSegment.java

```diff
@@ -235,55 +233,118 @@ void addInstanceSize(ClassDump cls, int tag, long instanceOffset) {
+                    
+                case "[F":
+                    // NOI18N
+                    typeObj = HprofHeap.FLOAT;
+                    break;
+                    
+                case "[D":
+                    // NOI18N
+                    typeObj = HprofHeap.DOUBLE;
+                    break;
+                    
+                case "[B":
+                    // NOI18N
+                    typeObj = HprofHeap.BYTE;
+                    break;
+                    
+                case "[S":
+                    // NOI18N
+                    typeObj = HprofHeap.SHORT;
+                    break;
+                    
+                case "[I":
+                    // NOI18N
+                    typeObj = HprofHeap.INT;
+                    break;
+                    
+                case "[J":
+                    // NOI18N
+                    typeObj = HprofHeap.LONG;
+                    break;
+                    
+                case "java/lang/Class":
+                    // NOI18N
+                    java_lang_Class = jcls;
+                    break;
+                    
+                case "java/lang/Object":
+                    // NOI18N
+                    java_lang_Object = jcls;
+                    break;
+                    
+                case "boolean[]":
+                    // NOI18N
+                    typeObj = HprofHeap.BOOLEAN;
+                    break;
+                    
+                case "char[]":
+                    // NOI18N
+                    typeObj = HprofHeap.CHAR;
+                    break;
+                    
+                case "float[]":
+                    // NOI18N
+                    typeObj = HprofHeap.FLOAT;
+                    break;
+                    
+                case "double[]":
+                    // NOI18N
+                    typeObj = HprofHeap.DOUBLE;
+                    break;
+                    
+                case "byte[]":
+                    // NOI18N
+                    typeObj = HprofHeap.BYTE;
+                    break;
+                    
+                case "short[]":
+                    // NOI18N
+                    typeObj = HprofHeap.SHORT;
+                    break;
+                    
+                case "int[]":
+                    // NOI18N
+                    typeObj = HprofHeap.INT;
+                    break;
+                    
+                case "long[]":
+                    // NOI18N
+                    typeObj = HprofHeap.LONG;
+                    break;
+                    
+                case "java.lang.Class":
+                    // NOI18N
+                    java_lang_Class = jcls;
+                    break;
+                    
+                case "java.lang.Object":
+                    // NOI18N
+                    java_lang_Object = jcls;
+                    break;
+                    
+                default:
+                    break;
```

Comment:
the intention is good but i think this makes it more difficult to read.

try to format it like:
```java
               case "[Z":  typeObj = HprofHeap.BOOLEAN;  break; // NOI18N
```
Its non-standard but would be compact and easier to read in this particular case. The "// NOI18N" hint must be on the same line as the String literal. 

Formatting it like this is also really close to switch expressions of later JDKs. Updating to them would only require the "->" and removal of "break".

But most importantly: we have to be careful with null here. What if vmName is null? The switch will throw an exception, the if-else would do fine.